crates/hyperswitch_connectors/src/connectors/nomupay.rs

```diff
@@ -0,0 +1,799 @@
+            }
+
+            // Ensure the parsed JSON is an object and convert it to `Map<String, Value>`
+            if let Value::Object(map) = value {
+                // Now you have a `Map<String, Value>`
+                println!("Parsed Map: {:?}", map);
+                Ok(map)
+            } else {
+                return Err(error)
+            }
+        },
+        _=> return Err(error)
+    }
+}
+*/
+
+fn get_private_key(
+    metadata: Option<pii::SecretSerdeValue>,
+) -> Result<String, errors::ConnectorError> {
+    match nomupay::NomupayMetadata::try_from(&metadata) {
+        Ok(nomupay_metadata) => Ok(nomupay_metadata.private_key),
+        Err(_e) => Err(errors::ConnectorError::NoConnectorMetaData),
+    }
+}
+
+fn box_to_jwt_payload(
+    body: Box<dyn masking::ErasedMaskSerialize + Send>,
+) -> Result<JwtPayload, errors::ConnectorError> {
+    let error = errors::ConnectorError::GenericError {
+        error_message: "josh error".to_string(),
+        error_object: Value::String("unable to create the body".to_string()),
+    };
+    // Step 1: Serialize `body` to JSON
+    let str_result = serde_json::to_string(&body);
+    let json_str;
+    if str_result.is_ok() {
+        json_str = str_result.map_err(|_e| errors::ConnectorError::GenericError {
+            error_message: "josh error".to_string(),
+            error_object: Value::String("unable to create the body".to_string()),
+        })?;
+
+        let map_result: Result<Map<String, Value>, serde_json::Error> =
+            serde_json::from_str(&json_str);
+        let parsed_json: Map<String, Value> = match map_result {
+            Ok(map) => map,
+            Err(_e) => {
+                return Err(errors::ConnectorError::GenericError {
+                    error_message: "josh error".to_string(),
+                    error_object: Value::String("unable to create the body".to_string()),
+                })
+            }
+        };
+
+        // Step 3: Use the `from_map` method to populate JwtPayload
+        let jwt_payload_result = JwtPayload::from_map(parsed_json);
+        if jwt_payload_result.is_ok() {
+            let jwt_payload = match jwt_payload_result {
+                Ok(payload) => payload,
+                Err(_e) => {
+                    return Err(errors::ConnectorError::GenericError {
+                        error_message: "josh error".to_string(),
+                        error_object: Value::String("unable to create the body".to_string()),
+                    })
+                }
+            };
+            return Ok(jwt_payload);
+        }
+    }
+    Err(error)
+}
+
+fn get_signature(
+    metadata: Option<pii::SecretSerdeValue>,
+    auth: nomupay::NomupayAuthType,
+    body: RequestContent,
+    method: String,
+    path: String,
+) -> Result<String, errors::ConnectorError> {
+    match body {
+        RequestContent::Json(masked_json) => {
+            let error = errors::ConnectorError::GenericError {
```

Comment:
Same here, let's not do this. Return the actual error from this function, and it can be mapped to ConnectorError in the caller function

Reply:
If there are various types of errors, we can map to ConnectorError at each instant